lib/rules/local-dependency.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @fileoverview Checks existence of local dependencies in the package.json
+ * @author Kendall Gassner
+ */
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const {
+    isPackageJson,
+    extractPackageObjectFromAST
+} = require('../processors/PackageJsonProcessor');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function fileExistsWithCaseSync(filepath) {
+    var dir = path.dirname(filepath);
+
+    if (dir === path.dirname(dir)) {
+        return true;
+    }
+    var filenames = fs.readdirSync(dir);
+    if (filenames.indexOf(path.basename(filepath)) === -1) {
+        return false;
+    }
+    return fileExistsWithCaseSync(dir);
+}
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                'Checks existence of local dependencies in the package.json',
+            category: 'Best Practices',
+            recommended: true
+        },
+        fixable: null // or "code" or "whitespace"
+    },
+
+    create: function(context) {
+        return {
+            'Program:exit': node => {
+                if (!isPackageJson(context.getFilename())) {
+                    return;
+                }
+                const sourceCode = context.getSourceCode();
+                const packageRoot = extractPackageObjectFromAST(node);
+                const original = JSON.parse(sourceCode.getText(packageRoot));
+                const {
+                    dependencies,
+                    peerDependencies,
+                    devDependencies
+                } = original;
+
+                const depObjs = [
+                    Object.entries(dependencies || {}),
+                    Object.entries(peerDependencies || {}),
+                    Object.entries(devDependencies || {})
+                ];
+
+                depObjs.map(obj =>
+                    obj.map(([key, value]) => {
+                        const response = localPath => {
+                            const filePath = path.join(
+                                context
+                                    .getFilename()
+                                    .replace(/package\.json/g, '/'),
```

Comment:
[Testing] Could you add a test case for the directory path containing `package.json`, such as a path like `path/to/package.json/gotcha/package.json`? It's a rare edge case, but not impossible.